src/Api.Rest/HttpClient/Data/DataServiceRestClient.cs

```diff
@@ -153,17 +153,22 @@ private async Task<IReadOnlyList<DataMeasurementDto>> GetMeasurementValuesSplitB
 			return result;
 		}
 
-		private static IReadOnlyCollection<DataCharacteristicDto> Combine( IReadOnlyCollection<DataCharacteristicDto> list1, IReadOnlyCollection<DataCharacteristicDto> list2 )
+		private static IReadOnlyDictionary<Guid, DataValueDto> Combine( IReadOnlyDictionary<Guid, DataValueDto> list1, IReadOnlyDictionary<Guid, DataValueDto> list2 )
 		{
 			if( list1 == null )
 				return list2;
 			if( list2 == null )
 				return list1;
 
-			var result = new List<DataCharacteristicDto>( list1.Count + list2.Count );
-
-			result.AddRange( list1 );
-			result.AddRange( list2 );
+			var result = new Dictionary<Guid, DataValueDto>( list1.Count + list2.Count );
+			foreach( var item in list1 )
+			{
+				result[ item.Key ] = item.Value;

```

Comment:
I think it would be a good idea to use TryAdd to throw a meaningful exception on duplicates in the dictionaries. If I remember correctly the old implementation would only throw on server side, but now we do get an exception when combining the dictionaries.

Reply:
Done. I avoid the special error message since the server should already return a consistent result and TryAdd is not part of the .Net Framework. I don't want to add extra work for performance reasons. However, by using Add instead of the indexer we get an exception if something is added twice.